frb_example/benches/dart/lib/interceptor.dart

```diff
@@ -0,0 +1,182 @@
+    final result = await super.executeNormal(task);
+    await interceptor.afterExecuteNormal(debugName, stopwatch);
+    return result;
+  }
+
+  @override
+  S executeSync<S>(FlutterRustBridgeSyncTask task) {
+    final String debugName = task.constMeta.debugName;
+    final TimeWatch stopwatch = interceptor.beforeExecuteSync(debugName, task.hint);
+    final result = super.executeSync(task);
+    interceptor.afterExecuteSync(debugName, stopwatch);
+    return result;
+  }
+
+  Future<List<Metric>?> metrics() async => interceptor is FlutterRustBridgeInterceptorJson
+      ? (interceptor as FlutterRustBridgeInterceptorJson).metrics.values.toList()
+      : null;
+}
+
+class FlutterRustBridgeExampleBenchmarkSuiteImplBench extends FlutterRustBridgeExampleBenchmarkSuiteImpl {
+  final FlutterRustBridgeExampleBenchmarkSuitePlatformBench platform;
+  factory FlutterRustBridgeExampleBenchmarkSuiteImplBench(ExternalLibrary dylib, {bool? useJSON}) =>
+      FlutterRustBridgeExampleBenchmarkSuiteImplBench.raw(
+          FlutterRustBridgeExampleBenchmarkSuitePlatformBench(dylib, useJSON ?? false));
+
+  /// Only valid on web/WASM platforms.
+  factory FlutterRustBridgeExampleBenchmarkSuiteImplBench.wasm(FutureOr<WasmModule> module, {bool? useJSON}) =>
+      FlutterRustBridgeExampleBenchmarkSuiteImplBench(module as ExternalLibrary, useJSON: useJSON ?? false);
+
+  @override
+  FlutterRustBridgeExampleBenchmarkSuiteImplBench.raw(this.platform) : super.raw((platform));
+
+  Future<List<Metric>?> dartMetrics() async {
+    return platform.metrics();
+  }
+}
+
+/// benchmark metric
+abstract class TimeWatch {
+  /// typically when the timer starts
+  int? starts;
+
+  /// typically when the timer ends
+  int? ends;
+
+  /// start timer
+  void start();
+
+  /// stop timer
+  void stop();
+
+  /// create a timer
+  TimeWatch.create();
+}
+
+/// unique benchmark metric
+abstract class UniqueTimeWatch extends TimeWatch {
+  /// allows for identifying same metric over time
+  ///
+  /// this way you can start a timer, save its value(s) somewhere,
+  /// and record whenever you stop it anytime later on.
+  late UuidValue uuid;
+  UniqueTimeWatch.create() : super.create();
+}
+
+abstract class FlutterRustBridgeInterceptor<T extends TimeWatch> {
+  T create();
+  Future<T> beforeExecuteNormal(String debugName, Object? hint) {
+    return Future.sync(() {
+      final T stopwatch = create();
+      stopwatch.start();
+      return stopwatch;
+    });
+  }
+
+  Future<void> afterExecuteNormal<S>(String debugName, T stopwatch) {
+    return Future.sync(() {
+      stopwatch.stop();
+    });
+  }
+
+  T beforeExecuteSync<S>(String debugName, Object? hint) {
+    final T stopwatch = create();
+    stopwatch.start();
+    return stopwatch;
+  }
+
+  void afterExecuteSync<S>(String debugName, T stopwatch) {
+    stopwatch.stop();
+  }
+}
+
+abstract class FlutterRustBridgeInterceptorStdOut<T extends TimeWatch> extends FlutterRustBridgeInterceptor<T> {
```

Comment:
IMHO, maybe we only need to support output JSON format, i.e. no need for directly logging here. If we want a stdout summary, we can parse the data that will become json and summarize it or format it.

Reply:
Well, it doesn't hurt to leave `stdout` mode, it's a good way to demonstrate a hybrid implementation and I've spent quite some time on it too. Why not just make `json` the default instead ?

Reply:
Good idea, but then the code quality may need a little bit of improvement :)